### PR TITLE
Use Processing for clipping

### PR DESCRIPTION
**Pull request by Gustry**

### CODE REVIEW

**Comment by ismailsunni** [comment]
LGTM

File: safe/impact_functions/base.py [hunk 1]
```diff
@@ -840,6 +884,65 @@ def _prepare(self):
...
+                                self.aggregation.qgis_layer().source(),
+                                self.hazard.qgis_layer().source(),
+                                self.exposure.qgis_layer().source(),
+                                exposure_result,
+                                hazard_result
+                            ]
+                        },
+                    }
+
+                    processing_if = processing_models[impact_function_id]
```
Comment (timlinux) [inline safe/impact_functions/base.py]: processing_if -> model

File: safe/impact_functions/inundation/flood_vector_building_impact/impact_function.py [hunk 2]
```diff
@@ -175,19 +161,15 @@ def run(self):
...
 
-        # Filter out just those EXPOSURE features in the analysis extents
-        transform = QgsCoordinateTransform(
-            self.requested_extent_crs, self.exposure.layer.crs())
-        projected_extent = transform.transformBoundingBox(requested_extent)
-        request = QgsFeatureRequest()
-        request.setFilterRect(projected_extent)
-
+        request = None
+        if not self.aggregation:
```
Comment (timlinux) [inline safe/impact_functions/inundation/flood_vector_building_impact/impact_function.py]: can remove this

File: safe/impact_functions/base.py [hunk 3]
```diff
@@ -654,6 +665,39 @@ def console_progress_callback(current, maximum, message=None):
             print message
         print 'Task progress: %i of %i' % (current, maximum)
 
+    def exposure_filter_request(self):
+        """Get the exposure filter request specific to the requested extent.
+
+        :return: The qgis feature request for the exposure layer.
```
Comment (timlinux) [inline safe/impact_functions/base.py]: QGIS

**Comment by timlinux** [comment]
Lets move this to 3.6 as I am worried it may introduce new regressions while we want to focus on spit & polish for 3.5

**Comment by Gustry** [comment]
Not valid anymore with inasafe v4

**Pull request closed without merging**